### ISSUE TITLE
duplicate cni0 and docker0

### ISSUE THREAD

**Issue by 372046933** [flannel-io/flannel#1013]
## Expected Behavior

With the default config `https://raw.githubusercontent.com/coreos/flannel/master/Documentation/kube-flannel.yml`, only the route table of the system should be altered.
## Current Behavior

`cni0` and `docker0` both appeared with the same IP. And the route table have the same route to these devices which are in conflict.
## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Your Environment

* Flannel version: 0.10
* Backend used (e.g. vxlan or udp): host-gw
* Etcd version: 3.2
* Kubernetes version (if used): 1.10
* Operating System and version: CentOS 7.4
* Link to your project (optional):

**Comment by ujnzxw** [comment]
I encountered same issue with you. How did you solve this?

**Comment by toldjuuso** [comment]
I ran to the same problem where bare-metal CoreOS Kubernetes installation control-plane node had the same IP for both `cni0` and `docker0`, which caused `no route to host` with pod-to-pod and pod-to-apiserver connections. I was able to solve the problem by following [a tangential issue](https://github.com/kubernetes/kubernetes/issues/39557), which [advised](https://github.com/kubernetes/kubernetes/issues/39557#issuecomment-457839765) to refresh the cni subnet by running `ip link delete cni0`. After this I restarted `kubelet` and everything started working.
